connectors/wsclient-javax.websocket/src/test/java/quarks/tests/connectors/wsclient/javax/websocket/WebSocketClientTest.java

```diff
@@ -678,6 +681,24 @@ public void testSslClientAuthNeg() throws Exception {
         completeAndValidate("", t, rcvd, SEC_TMO, new String[0]);  //rcv nothing
     }
     
+    private void skipTestIfCantConnect(Properties config) throws Exception {
+        String wsUri = config.getProperty("ws.uri");
+        // Skip tests if the WebSocket server can't be contacted.
+        try {
+            URI uri = new URI(wsUri);
+            int port = uri.getPort();
+            if (port == -1)
+                port = uri.getScheme().equals("ws") ? 80 : 443;
+            Socket s = new Socket();
+            s.connect(new InetSocketAddress(uri.getHost(), port), 5*1000/*cn-timeout-msec*/);
+            s.close();
+        } catch (Exception e) {
+            System.err.println("Unable to connect to WebSocket server "+wsUri+" : "+e.getMessage());
+            e.printStackTrace();
+            assumeTrue(false);
```

Comment:
Is it possible to use the overload that takes a message that will be reported in the junit report, or does that require a newer version of JUnit?